backend/v3/domain/instance_domain_remove.go

```diff
@@ -0,0 +1,119 @@
+package domain
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/zitadel/zitadel/backend/v3/storage/database"
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/repository/instance"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+type RemoveInstanceDomainCommand struct {
+	InstanceID string     `json:"instance_id"`
+	DomainName string     `json:"domain_name"`
+	DType      DomainType `json:"domain_type"`
+}
+
+// RequiresTransaction implements [Transactional].
+func (r *RemoveInstanceDomainCommand) RequiresTransaction() {}
+
+func NewRemoveInstanceDomainCommand(instanceID, domainName string, domainType DomainType) *RemoveInstanceDomainCommand {
+	return &RemoveInstanceDomainCommand{
+		InstanceID: instanceID,
+		DomainName: domainName,
+		DType:      domainType,
+	}
+}
+
+// Events implements [Commander].
+func (r *RemoveInstanceDomainCommand) Events(ctx context.Context, _ *InvokeOpts) ([]eventstore.Command, error) {
+	var toReturn eventstore.Command
+	switch r.DType {
+	case DomainTypeCustom:
+		toReturn = instance.NewDomainRemovedEvent(ctx, &instance.NewAggregate(r.InstanceID).Aggregate, r.DomainName)
+	case DomainTypeTrusted:
+		toReturn = instance.NewTrustedDomainRemovedEvent(ctx, &instance.NewAggregate(r.InstanceID).Aggregate, r.DomainName)
+	default:
+		return nil, nil
+	}
+	return []eventstore.Command{toReturn}, nil
+}
+
+// Execute implements [Commander].
+func (r *RemoveInstanceDomainCommand) Execute(ctx context.Context, opts *InvokeOpts) (err error) {
+	instanceRepo := opts.instanceDomainRepo
+	// TODO(IAM-Marco): I'm not sure if this is a mistake, but InstanceIDCondition must be passed, otherwise an error is returned.
```

Comment:
This comes form the restriction under `if !condition.IsRestrictingColumn(i.InstanceIDColumn()) {`.